services/sentry/lib/eventReducer.js

```diff
@@ -56,22 +64,41 @@ function isPriceOk(price, channel) {
 
 	return price.gte(minPrice) && price.lte(maxPrice)
 }
+// returns the price per impression
+function getPrice(channel, ev) {
+	let price = new BN(channel.currentPricePerImpression || channel.spec.minPerImpression, 10)
+	if (channel.pricePerImpressionCase && ev.stat) {
+		// check if there is a unique price case
+		// for the event stat
+		// it can be publisher prefixed for a specific publisher
+		const result = channel.pricePerImpressionCase.find(
+			priceCase => priceCase.stat === ev.stat || priceCase.stat === `${ev.publisher}:${ev.stat}`
```

Comment:
this allows setting the price per-publisher, right?

Reply:
Yes